### DISCUSSION TITLE
Bastion Host 만들기 강의에서 Private EC2 접속 문제

### DISCUSSION THREAD

**Discussion by dongdongdavid** [tedilabs/fastcampus-devops#13] [Q&A] [answered]
안녕하세요 현재 한번에 끝내는 AWS 인프라 구축과 Devops 운영을 듣고 있는 수강생입니다.

Part2를 여기에 질문해도 될지는 모르겠지만 Part2에 나오는 강사분이 질문할 곳을 올려주시지 않아서 이쪽에 질문드립니다.
강의대로 VPC 생성 후 아래와 같이 설정하였습니다.
testvpc IPv4 CIDR - 10.0.0.0/16
Private Subnet - 10.0.1.0/24
Public Subnet  - 10.0.0.0/24 DHCP 활성화

인터넷 게이트웨이 생성 후 testvpc에 연결하여 attached 상태 확인

라우팅테이블
private-rtb
서브넷 연결
-> 명시적서브넷연결없이 기본서브넷인 Private subnet으로 연결
라우팅
대상 10.0.0.0/16 local

라우팅 테이블
public rtb 
서브넷 연결
-> 명시적 서브넷 연결 - Public subnet 에 연결
라우팅
대상 10.0.0.0/16 local
대상 0.0.0.0/0     igw(testvpc에 연결한 인터넷게이트웨이)

네트워크 ACL  설정
기본 NACL 
private-nacl
연결대상 서브넷 private subnet
인바운드 / 아웃바운드 규칙 동일하게 아래와 같이 설정
규칙 번호 | 유형 | 프로토콜 | 포트범위 | 대상 | 허용/거부
100 | 모든 트래픽 | 모두 | 모두 | 0.0.0.0/0 | Allow
*| 모든 트래픽 | 모두 | 모두 | 0.0.0.0/0 | Deny

추가한NACL
public-nacl
연결대상 서브넷 public subnet
인바운드규칙 

규칙 번호 | 유형 | 프로토콜 | 포트범위 | 대상 | 허용/거부
100 | SSH(22) | TCP(6) | 22 | 0.0.0.0/0 | Allow
200 | HTTP(80) | TCP(6) | 80 | 0.0.0.0/0 | Allow
300 | HTTPS(443) | TCP(6) | 443 | 0.0.0.0/0 | Allow
*| 모든 트래픽 | 모두 | 모두 | 0.0.0.0/0 | Deny

아웃바운드규칙

규칙 번호 | 유형 | 프로토콜 | 포트범위 | 대상 | 허용/거부
100 | 사용자 지정 TCP | TCP(6) | 1024 - 65535 | 0.0.0.0/0 | Allow
200 | 모든 TCP | TCP(6) | 모두 | 10.0.1.0/24 | Allow 
// 200번 옵션은 원래 없었는데 하두 안되서 추가해보았습니다. 이거 추가해도안됩니다.,.,,,,,,,,,
-- | -- | -- | -- | -- | --
*| 모든 트래픽 | 모두 | 모두 | 0.0.0.0/0 | Deny

EC2 정보
public-ec2 (배스천호스트)
testvpc, public subnet 바라보도록 설정
보안그룹 설정
public-sg
인바운드 설정
IP version | Type | Protocol | Port range | Source 
ipv4 |모든 icmp-ipv4 | ICMP  | 전체 | 0.0.0.0/0
ipv4 | SSH | ICMP  | 22 | 0.0.0.0/0

아웃바운드 설정
IP version | Type | Protocol | Port range | Source 
ipv4 |모든 트래픽 | 전체  | 전체 | 0.0.0.0/0

Private EC2 정보
private-ec2
testvpc, private subnet 바라보도록 설정
보안그룹 설정
private-sg
인바운드 설정
IP version | Type | Protocol | Port range | Source 
ipv4 |모든 icmp-ipv4 | ICMP  | 전체 | public-sg
ipv4 | SSH | ICMP  | 22 | public-sg

아웃바운드 설정
IP version | Type | Protocol | Port range | Source 
ipv4 |모든 트래픽 | 전체  | 전체 | 0.0.0.0/0

위와 같이 진행 후 EC2를 각각 다른 키페어로 만들고 Pagent에 각 pem을 ppk로 변환한 파일을 업로드 한뒤
Putty 설정에서는 user ec2-user / SSH-Auth 옵션에 Allow agent forwarding 체크 후 접속했음에도
public ec2는 접근이 되지만 거기에서 private ec2는 접근이 안됩니다.

서브넷은 두개가 같은 리전에 속해있습니다.
public ec2에서 private ec2로 ping도 안갈 뿐더러 강의대로 몇번을 다시만들고 시도해봐도 되질 않아서 
접근이 안되는 이유가 제가 어떤 부분이 잘못 설정했는지 확인 부탁드립니다.

P.S 강의에서 라우팅을 루트로 자꾸 발음하셔서 처음에는 조금 당황했습니다ㅠㅠ

**Comment by posquit0** [comment, marked as answer]
안녕하세요~
해당 강사님 질문은 패스트캠퍼스 측에 이메일로 문의를 하면 강사님께 포워딩되어 질문응답이 이루어집니다. :)

질문 내용을 자세히 살펴보지는 못하였지만, private-ec2에 SSH를 접근할 때 private-ip를 사용하였는지 public-ip를 사용하였는지 체크해볼 필요가 있어보입니다.

**Reply by dongdongdavid** [reply]
패스트캠퍼스측에 문의하니 그대로했음에도 안되는건데 반복수강하라는답변만 주시더라구요..ㅎㅎㅎㅎ
당연히 예를들어 private-ec2가 private-ip는 10.0.1.x/32라서 10.0.0.x 인 퍼블릭에서 접근시도를 ssh 10.0.1.x 로도 ssh ec2-user@10.0.1.x로도 했는데 접근이 안되더라구요 ㅜㅜ 해결되는건없지만 그래도 답변주셔서 감사합니다.

**Reply by posquit0** [reply]
public nacl의 inbound rule에 400 | 사용자 지정 TCP | TCP(6) | 1024 - 65535 | 0.0.0.0/0 | Allow 넣고 한 번 다시 해보시겠어요?

**Reply by dongdongdavid** [reply]
아... 인바운드 룰에 넣으니 되네요ㅠ  퍼블릭에서 private으로 22 포트로 연결 시도를 하면 private에서 다시 퍼블릭으로 연결이 맺어질때 tcp 응답을 1024-65535로 주는것 때문에 nacl에 해당 포트를 넣어줘야하는군요ㅠ 도움주셔서 정말 감사합니다!

**Reply by posquit0** [reply]
ㅎㅎ 다행입니다.
